### PR TITLE
chore: simplify digest boundary logic

### PR DESCRIPTION
**Pull request by MicBun**
review after: https://github.com/trufnetwork/node/pull/1123

Implement simplified day boundary approach with exclusive end boundaries
and correct deletion counting logic in batch digest operations.

**Key Changes:**
- Replace complex boundary conflict detection with standard exclusive 
  intervals (`< day_end` instead of `<= day_end`)
- Fix SQL deletion counting logic to report accurate deletion counts 
  instead of always showing 0
- Enable single-record stream processing with >= 1 condition
- Remove 100+ lines of over-engineered boundary detection code
- Update test data to work with exclusive day boundaries

**Impact:**
- Eliminates timestamp collision bugs that caused type flag conflicts
- Provides accurate deletion metrics for monitoring and debugging
- Improves performance by removing expensive LAG/LEAD window functions
- Enhances maintainability with standard interval approach
- Ensures reliable OHLC data processing for financial metrics

**Technical Details:**
- Changed the SQL boundary conditions from inclusive to exclusive
- Moved deletion counting before DELETE operation to capture accurate counts
- Updated boundary test data (172800→172799, 259200→259199, etc.)
- All test scenarios pass with correct deletion reporting

Fixes boundary overlap issues while maintaining OHLC calculation accuracy.

resolves: https://github.com/trufnetwork/node/issues/1127

## Summary by CodeRabbit

* **New Features**
  * Added batch digest to process multiple streams/days at once for improved performance and scalability.
  * Auto-digest now batches candidates and returns processed_days, total_deleted_rows, and total_preserved_rows.

* **Refactor**
  * Replaced per-day digest flow with a bulk batch approach; per-day action removed and input/return shapes updated.

* **Tests**
  * Extended and replaced digest tests to cover batch scenarios, ordering, deletion logic, and edge cases.

* **Chores**
  * Updated core dependencies to newer builds.

### CODE REVIEW

File: internal/migrations/020-digest-actions.sql [hunk 1]
```diff
@@ -1,231 +1,617 @@
...
+                ORDER BY pe.stream_ref, pe.event_time, pe.created_at
+                LIMIT $delete_cap
+            )
+            SELECT 
+                (SELECT COUNT(*) FROM delete_candidates) as deleted_count,
+                (SELECT COUNT(*) FROM keep_set) as preserved_count {
+                
+                $total_deleted := $delete_count_row.deleted_count;
+                $total_preserved := $delete_count_row.preserved_count;
+            }
```
[the remark this answers is not in the document]
Comment (MicBun) [reply internal/migrations/020-digest-actions.sql:604]: Unfortunately RETURNING syntax is not supported by kwil-db

File: internal/migrations/020-digest-actions.sql [hunk 2]
```diff
@@ -1,231 +1,617 @@
...
+                   ((u.day_index * 86400) + 86400) as day_end
+            FROM UNNEST($valid_stream_refs, $valid_day_indexes) AS u(stream_ref, day_index)
+        ),
+        stream_record_counts AS (
+            -- Count records for each stream in single query
+            SELECT sd.stream_ref, sd.day_index, sd.day_start, sd.day_end,
+                   COUNT(pe.event_time) as record_count
+            FROM stream_days sd
+            LEFT JOIN primitive_events pe ON pe.stream_ref = sd.stream_ref
+                AND pe.event_time >= sd.day_start AND pe.event_time <= sd.day_end
```
Comment (outerlook) [inline internal/migrations/020-digest-actions.sql:107]: hey @MicBun, to solve this very issue, isn't better to define the day as being not inclusive between the two intervals?

as in:

| day | day_start | day_end |
| 1   | 0         | 86399   |
| 2   | 86400     | 172799  |

then the query would be:

```suggestion
                AND pe.event_time >= sd.day_start AND pe.event_time < sd.day_end
```

my concern here is about having the same value on both dates, or conditioned to having conflict or not to be part of one day or the other

or did I get this issue wrong?

Comment (MicBun) [reply internal/migrations/020-digest-actions.sql:107]: How should we define day_start and day_end? What's our strategy for handling conflicting values?

While using `<` offers faster implementation, consider this edge case scenario:

**Stream A (hourly ingestion):**
```
82800 | 80
86400 | 100
```
With hourly ingestion, it appears the day closes at 80 and opens the next day at 100.

**However, in real-time scenarios:**
```
82800 | 80
...
86399 | 99.99
86400 | 100
```
This presents a specific edge case where the apparent day-end value differs significantly between batch and real-time processing.

The resolution depends on how we define day_end boundaries. Would it be more efficient to establish clear day_end definitions upfront rather than adding conditional logic to handle date conflicts? Anyway, I'm fine with changing it to `<` instead of `<=` if we can just ignore this edge case. What approach do you think would be most robust? @outerlook
